src/bgw_policy/policy.c

```diff
@@ -36,18 +37,38 @@ ts_bgw_policy_delete_row_only_tuple_found(TupleInfo *ti, void *const data)
 	return SCAN_CONTINUE;
 }
 
-int32
-ts_bgw_policy_reorder_count()
+/* This function counts background worker jobs by type. */
+BgwJobTypeCount
+ts_bgw_job_type_counts()
 {
-	List *jobs = ts_bgw_job_find_by_proc("policy_reorder", INTERNAL_SCHEMA_NAME);
+	ListCell *lc;
+	List *jobs = ts_bgw_job_get_all(sizeof(BgwJob), CurrentMemoryContext);
+	BgwJobTypeCount counts = { 0 };
 
-	return list_length(jobs);
-}
+	foreach (lc, jobs)
+	{
+		BgwJob *job = lfirst(lc);
 
-int32
-ts_bgw_policy_retention_count()
-{
-	List *jobs = ts_bgw_job_find_by_proc("policy_retention", INTERNAL_SCHEMA_NAME);
+		if (namestrcmp(&job->fd.proc_schema, INTERNAL_SCHEMA_NAME) == 0)
+		{
+			if (namestrcmp(&job->fd.proc_name, "policy_refresh_continuous_aggregate") == 0)
```

Comment:
I would prefer that this isn't hard coded with free-floating string  values. This will be cumbersome to maintain when we add new policies and it would be better to either use a hash table to collect policy counts, or use a static list of officially supported policies that we can reference throughout the code.

I guess this is a bigger refactoring, though, and not for this PR.